### ISSUE TITLE
Tkinter import error for pyenv Pythons

### ISSUE THREAD

**Issue by perscitus** [pyenv/pyenv#94]
Debian running system Python (2.7.3):

```
python
import Tkinter
```

Works just fine.

```
pyenv install 2.7.6
pyenv global 2.7.6
python
import Tkinter
```

Fails with ImportError: No module named _tkinter. Same issue occurs with Python 3.3.3 with pyenv. Tried clearing ~/.pyenv/cache. Also tried `pyenv uninstall 2.7.6` and `pyenv install 2.7.6` afterwards.

I'm guessing `pyenv install` is not building python with tkinter at compile time?

**Comment by blueyed** [comment]
Thanks @perscitus!

It helped me and I've answered a SO question with it: http://stackoverflow.com/a/31299142/15690

**Comment by robsonpeixoto** [comment]
@yyuu Do you know how o make it works on OSX ?

**Comment by lezuber** [comment]
TL;DR On OSX use 'brew install tcl-tk' before installing any CPython Version with pyenv
This should also be se solution for @robsonpeixoto 

A similar strange problem took me way to long today to solve.
I have installed some CPython versions with pyenv, and was able to import tkinter both in python 3 and 2. However it was not properly working. In many cases it acted very strange, and only blank windows were opening. Maybe it should be described in the docs, to install tcl-tk, before using pyenv. 

using 'brew install tcl-tk' before installing python versions solved all my problems on OSX

**Comment by kuronekokami** [comment]
i am having this problem on solus os, even with tk-devel package installed. i am able to import the tkinter module on my system pythons but cannot do so on my pyenv install. 

'import tkinter' and 'import _tkinter' both fail.

 i have made sure that i installed the tk-devel package before installing pyenv.

**Comment by carlgieringer** [comment]
On MacOS you may need to set some env. vars. in order for the pyenv installation to pick up the homebrew tcl-tk:

```
※ pyenv uninstall 3.8.1
※ export PATH="/usr/local/opt/tcl-tk/bin:$PATH"
※ export LDFLAGS="-L/usr/local/opt/tcl-tk/lib"
※ export CPPFLAGS="-I/usr/local/opt/tcl-tk/include"
※ export PKG_CONFIG_PATH="/usr/local/opt/tcl-tk/lib/pkgconfig"
※ pyenv install 3.8.1
```

[SO Q/A](https://stackoverflow.com/a/60469203/39396)

**Comment by dggsax** [comment]
@carlgieringer Your solution worked for me, thank you! Should the solution you provided be offered as a flag for `pyenv install`? Such as `pyenv install 3.7.6 --with-tkinter`? Or is your solution more of a one stop shop since as far as it appears the exports are temporary.

**Comment by schilli91** [comment]
@carlgieringer Your instructions at least solved the problem that I couldn't import tkinter in a pyhton shell. However, when I try running my script, which is using matplotlib (that is requiring tkinter), I receive the following issue:

RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)

So far I couldn't find a simple solution for that. Do you have any advice?

**Comment by unlessbamboo** [comment]
OS: ubuntu
Python virtualenv versions:3.11.5
Command: sudo apt-get install tk-dev
This issue was solved via link:https://dev.to/xshapira/using-tkinter-with-pyenv-a-simple-two-step-guide-hh5

**Comment by sjlawson** [comment]
> `pyenv` doesn't manage dependencies relating to `Tkinter`.
> 
> It seems that the dependencies of `Tkinter` has not been installed on your system. If you're using Debian, please try `sudo apt-get build-dep python-tk`. Then, rebuilding CPython will resolve your issue.

OS: Debian 12 - this issue is still affecting pyenv on Debian 12. The apt-get solution above is *almost* what is required.
I needed this:
`sudo apt-get build-dep python3-tk` 

Now it seems good.

**Comment by teddylegessemunea** [comment]
This is the problem of not configuring your Python with tcltk

1. Install Tk development libraries as mentioned above:

```
sudo apt update
sudo apt install -y tk-dev libncurses5-dev libncursesw5-dev libreadline-dev libdb5.3-dev libgdbm-dev libsqlite3-dev libssl-dev libbz2-dev libexpat1-dev liblzma-dev zlib1g-dev uuid-dev
```

2. Download Python source you want to install (I am using python3.11.9)

```
cd /usr/src
sudo wget https://www.python.org/ftp/python/3.11.9/Python-3.11.9.tgz
sudo tar xvf Python-3.11.9.tgz
cd Python-3.11.9
```

3. Configure the downloaded python with Tcl/Tk enabled

`sudo ./configure --enable-optimizations --with-tcltk`

4. compile it

`sudo make -j$(nproc)`

5. Finally, install using command altinstall (to avoid overwriting the system’s default python3)

`sudo make altinstall`

I tried my code again and this worked for me
